pkg/api/defaults.go

```diff
@@ -438,16 +438,20 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool) {
 			profile.AcceleratedNetworkingEnabledWindows = helpers.PointerToBool(DefaultAcceleratedNetworkingWindowsEnabled)
 		}
 
+		// Ensure distro is set properly for N Series SKUs no matter what, because
+		// (1) At present, "aks-docker-engine" is the only working distro base for running GPU workloads on N Series SKUs
```

Comment:
that's not technically true, "Ubuntu" still works. Does this mean agent pool profiles with the Ubuntu distro will be forced changed to AKSDockerEngine? What if I want to request Ubuntu as a user?

Reply:
```
if profile.OSDiskSizeGB != 0 && profile.OSDiskSizeGB < VHDDiskSizeAKS {
					profile.Distro = Ubuntu
				}
```
should still apply as well